sycl/test/basic_tests/range_offset_fit_in_int.cpp

```diff
@@ -0,0 +1,178 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -D__SYCL_ID_QUERIES_FIT_IN_INT__=1 %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <climits>
+
+namespace S = cl::sycl;
+
+void checkRangeException(S::runtime_error &E) {
+  constexpr char Msg[] = "Provided range is out of integer limits. Suggest "
+                         "disabling `id-queries-fit-in-int32' optimizations "
+                         "flag.";
+
+  std::cerr << E.what() << std::endl;
+
+  assert(std::string(E.what()).find(Msg) == 0 && "Unexpected message");
+}
+
+void checkOffsetException(S::runtime_error &E) {
+  constexpr char Msg[] = "Provided offset is out of integer limits. Suggest "
+                         "disabling `id-queries-fit-in-int32' optimizations "
+                         "flag.";
+
+  std::cerr << E.what() << std::endl;
+
+  assert(std::string(E.what()).find(Msg) == 0 && "Unexpected message");
+}
+
+void test() {
+  auto EH = [](S::exception_list EL) {
+    for (const std::exception_ptr &E : EL) {
+      throw E;
+    }
+  };
+
+  S::queue Queue(EH);
+
+  static constexpr size_t OutOfLimitsSize = static_cast<size_t>(INT_MAX) + 1;
+
+  S::range<1> RangeOutOfLimits{OutOfLimitsSize};
+  S::range<1> RangeInLimits{1};
+  S::id<1> OffsetOutOfLimits{OutOfLimitsSize};
+  S::id<1> OffsetInLimits{1};
+  S::nd_range<1> NDRangeROLLILOIL{RangeOutOfLimits, RangeInLimits,
```

Comment:
These are quite hard to read. Maybe putting underscores helps?
```suggestion
  S::nd_range<1> NDRangeROL_LIL_OIL{RangeOutOfLimits, RangeInLimits,
```